level1c4pps/vgac2pps_lib.py

```diff
@@ -96,58 +98,262 @@
+                "lutning": 0.9934,
+                "offset": 1.52,
+                "comment": "based on nadir collocation data for SZA 0-180",
+            },
+        },
+        "v6": {
+            "r06": {
+                "VIIRS channel": "M05",
+                "lutning": 0.9393,
+                "offset": 0.57,
+                "comment": "Based on collocation data, VZA < 3 for SZA < 60",
+            },
+            "r09": {
+                "VIIRS channel": "M07",
+                "lutning": 0.9104,
+                "offset": -0.90,
+                "comment": "Based on collocation data, VZA < 3 for SZA < 60",
+            },
+            "tb37_night": {
+                "VIIRS channel": "M12",
+                "lutning": 0.9967,
+                "offset": 0.85,
+                "rule": np.array(scene["sunzenith"].values) >= 89.,
+                "comment": "Based on collocations data for SZA >= 89",
+            },
+            "tb37_day": {
+                "VIIRS channel": "M12",
+                "lutning": 0.9455,
+                "offset": 12.69,
+                "rule": np.array(scene["sunzenith"].values) < 80.,
+                "comment": "Based on collocations for SZA < 80 o and VZA < 5",
+            },
+            "tb37_twilight": {
+                "VIIRS channel": "M12",
+                "lutning": 0.9711,
+                "offset": 6.77,
+                "rule": np.array((scene["sunzenith"].values >= 80) & (scene["sunzenith"].values < 89.)),
+                "comment": "The linear average of the SBAFs for t37_day and t37_night. 80<= SZA <89",
+            },
+            "tb11": {
+                "VIIRS channel": "M15",
+                "lutning": 1.003,
+                "offset": -0.77,
+                "comment": "Based on collocation data for VZA < 3 and SZA 0 -180",
+            },
+            "tb12": {
+                "VIIRS channel": "M16",
+                "lutning": 1.002,
+                "offset": -0.69,
+                "comment": "Based on collocation data for VZA < 3 and SZA 0-180",
+            },
+        }
+    }
+
+    strlen = max(len(k) for k in SBAF[V].keys())
+    tb37_separated = ['tb37_day', 'tb37_night', 'tb37_twilight']
+    print(f"Using SBAF_{V}")
+    for y in SBAF[V].keys():
+        k = SBAF[V][y]['lutning']
+        x = SBAF[V][y]['VIIRS channel']
+        m = SBAF[V][y]['offset']
+        com = SBAF[V][y]['comment']
+        if y not in tb37_separated:
+            print(f"{y:<{strlen}} = {k:<6}*{x:<3}+{m:<5} ({com})")
+            scene[x].values = k * scene[x].values + m
+
+    # # #####  Ch3.7 special handling
+    if any(y in SBAF[V] for y in tb37_separated):
+        M12 = np.array(scene["M12"].values)
+        for y in tb37_separated:
+            k = SBAF[V][y]['lutning']
+            m = SBAF[V][y]['offset']
+            rule = SBAF[V][y]['rule']
+            com = SBAF[V][y]['comment']
+
+            print(f"{y:<{strlen}} = {k:<6}*M12+{m:<5} ({com})")
```

Comment:
I think it would be better to add a unit test for this function instead of having these prints